pilot/pkg/model/service.go

```diff
@@ -1312,13 +1310,31 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 	return s.getAllAddressesForProxy(node)
 }
 
+// nodeUsesAutoallocatedIPs checks to see if this node is eligible to consume automatically allocated IPs
+func nodeUsesAutoallocatedIPs(node *Proxy) bool {
+	if node == nil {
+		return false
+	}
+	var DNSAutoAllocate, DNSCapture bool
+	if node.Metadata != nil {
+		DNSAutoAllocate = bool(node.Metadata.DNSAutoAllocate)
+		DNSCapture = bool(node.Metadata.DNSCapture)
+	}
+	// check whether either version of auto-allocation is enabled
+	autoallocationEnabled := DNSAutoAllocate || features.EnableIPAutoallocate
+
+	// check if this proxy is a type that always consumes or has consumption explicitly enabled
+	nodeConsumesAutoIP := DNSCapture || node.Type == Waypoint
+
+	return autoallocationEnabled && nodeConsumesAutoIP
```

Comment:
Based on the issue description:
> waypoint proxies do not get auto-IPs configured unless ISTIO_META_DNS_CAPTURE is enabled but the desired behavior should be that waypoints always use auto-IPs.

Shouldn't we automatically return true if `node.Type == Waypoint` without the additional condition that autoallocation also be set to true? Or is it the case that Waypoints _always_ have autoallocation true in this context? So, technically, `autoallocationEnabled && nodeConsumesAutoIP` is always true when referring to Waypoints?

Reply:
It should have been stated as "waypoints should not require DNScapture to be enabled", but it was stated in an unclear way. I think having some form of auto-IP enabled in istiod does still make sense as a prereq for waypoints. I suppose the argument could be made that waypoints are for Ambient so they could ignore the version of autoIP that isn't compatible with Ambient though. Unclear to me if there is a future where sidecars are using the old feature to route traffic to like an egress waypoint that would then benefit from the listener being there even though ambient auto-IP isn't running...

Reply:
Thanks for clarifying. Makes sense now. Good work!